pkg/api/apiv1/checkpoint.go

```diff
@@ -430,6 +492,40 @@ func (a checkpointAPI) CheckpointResponse(w http.ResponseWriter, r *http.Request
 	}
 }
 
+func (a checkpointAPI) Output(w http.ResponseWriter, r *http.Request) {
+	// Assert that we have a checkpoint JWT in the query param.
+	token := r.URL.Query().Get("token")
+
+	claims, err := apiv1auth.VerifyRunJWT(r.Context(), a.runClaimsSecret, token)
+	if err != nil || claims == nil {
+		w.WriteHeader(401)
+		_ = publicerr.WriteHTTP(w, publicerr.Wrap(err, 401, "Unable to find run with auth token"))
+		return
+	}
+
+	if a.outputReader == nil {
+		w.Header().Set("content-type", "application/json")
+		_, _ = w.Write([]byte(`{"status":"unknown","message":"unable to fetch run output"}`))
+		return
+	}
```

Comment:
Can we log something internally here (or `reportError()`) as this indicates bad setup somewhere?